src/search/migration/import-export.test.ts

```diff
@@ -0,0 +1,159 @@
+            index.addTag(data.PAGE_DOC_1.url, tag),
+        ),
+    )
+    await index.addBookmark({
+        url: data.PAGE_DOC_1.url,
+        timestamp: data.TEST_BOOKMARK_1,
+    } as any)
+    await addPouchPageAttachment(
+        data.PAGE_DOC_1._id,
+        'screenshot',
+        data.TEST_SCREENSHOT,
+    )
+    await addPouchPageAttachment(
+        data.PAGE_DOC_1._id,
+        'favIcon',
+        data.TEST_FAVICON,
+    )
+}
+
+async function resetDataSources(dbName = 'test') {
+    // Don't have any destroy methods available;
+    //   => update pointer to memdown and manually delete fake-indexeddb's DB
+    indexedDB.deleteDatabase(dbName)
+    oldIndex.init({ levelDown: memdown() })
+    newIndex.init({ indexedDB, IDBKeyRange, dbName })
+}
+
+describe('Old=>New index migration', () => {
+    describe('read ops', () => {
+        beforeAll(async () => {
+            await resetDataSources()
+            await insertTestPageIntoOldIndex()
+        })
+
+        test('Exporting old-index data', async () => {
+            for await (const { pages: [page] } of exportOldPages()) {
+                expect(page).toEqual(data.EXPORTED_PAGE_1)
+            }
+        })
+    })
+
+    describe('read-write ops', () => {
+        beforeEach(async () => {
+            await resetDataSources()
+            await insertTestPageIntoOldIndex()
+        })
+
+        // Try to find the page data stored for given test data from DB - check everything
+        async function testStoredPage(expectedData: Partial<ExportedPage>) {
+            const storedPage = await newIndex.default.pages.get(
+                expectedData.url,
+            )
+
+            const {
+                visits,
+                screenshotURI,
+                favIconURI,
+                bookmark,
+                tags,
+                ...expected,
+            } = expectedData
+            const { screenshot, ...page } = storedPage
+
+            // Test standard data
+            expect(page).toEqual(expected)
+
+            // Test assoc. data via getters
+            await storedPage.loadRels()
+            expect(storedPage.screenshotURI).toBe(screenshotURI)
+            expect(storedPage.hasBookmark).toBe(true)
+            expect(storedPage.bookmark.time).toBe(bookmark)
+
+            expect(storedPage.tags).toEqual(expect.arrayContaining(tags))
+            expect(storedPage.visits.map(visit => visit.time)).toEqual(
+                expect.arrayContaining(visits.map(visit => visit.time)),
+            )
+
+            await newIndex.default.favIcons
+                .get(expectedData.hostname)
+                .then(storedFav => {
+                    if (storedFav) {
+                        expect(storedFav.hostname).toBe(page.hostname)
+                        expect(storedFav.favIconURI).toBe(favIconURI)
+                    }
+                })
+        }
+
+        test('Importing data to new index', async () => {
+            index.useOld = false
```

Comment:
When I'm importing my data, isn't useOld still set to true?

Reply:
Yes, actually the `import` code just uses Dexie directly now (to avoid a lot of unneeded processing our index interface methods, like `addNewPage`, do), so the `useOld` value is not used. This line is meant for the following `index.search` call, so the search happens on the freshly migrated new index. Very confusing here, so will move it.